pkg/util/version_history.go

```diff
@@ -0,0 +1,83 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package util
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+type versionHistoryEntry struct {
+	Version   string    `json:"version"`
+	Timestamp time.Time `json:"timestamp"`
+}
+
+type versionHistoryEntries struct {
+	Entries []versionHistoryEntry `json:"entries"`
+}
+
+const maxVersionHistoryEntries = 60
+
+// LogVersionHistory loads version history file, append new entry if agent version is different than the last entry in the
+// JSON file, trim the file if too many entries then save the file.
+func LogVersionHistory() {
+	versionHistoryFilePath := filepath.Join(config.Datadog.GetString("logs_config.run_path"), "version-history.json")
+	logVersionHistoryToFile(versionHistoryFilePath, version.AgentVersion, time.Now().UTC())
```

Comment:
I think you can leave out the second and third parameters `version.AgentVersion` and `time.Now().UTC()` and do those things inside the function, or what would be the benefit of doing it like this?

Reply:
How would you do test in that case???

Reply:
Yeah you're right, makes sense from a testability point of view.